luxonis_ml/nn_archive/config_building_blocks/base_models/head.py

```diff
@@ -190,93 +148,146 @@ def validate_label_type(
+        OutputsKeypointDetectionYOLO,
+        OutputsOBBDetectionYOLO,
+    ] = Field(
         description="A configuration specifying which output names from the `outputs` block of the archive are fed into the head."
     )
-    postprocessor_path: str = Field(
-        ...,
+    subtype: ObjectDetectionSubtypeYOLO = Field(
+        description="YOLO family decoding subtype (e.g. v5, v6, v7 etc.)."
+    )
+    postprocessor_path: Optional[str] = Field(
+        None,
         description="Path to the secondary executable used in YOLO instance segmentation.",
     )
-    n_prototypes: int = Field(description="Number of prototypes per bbox.")
-
-    @field_validator("family")
-    def validate_label_type(
-        cls,
-        value,
-    ):
-        if value != "InstanceSegmentationYOLO":
-            raise ValueError("Invalid family")
-        return value
-
-
-class HeadKeypointDetectionYOLO(HeadObjectDetectionYOLO, ABC):
-    """Metadata for YOLO keypoint detection head.
-
-    @type family: str
-    @ivar family: Decoding family.
-    @type outputs: C{OutputsKeypointDetectionYOLO}
-    @ivar outputs: A configuration specifying which output names from the `outputs` block of the archive are fed into the head.
-    @type n_keypoints: int
-    @ivar n_keypoints: Number of keypoints per bbox.
-    """
-
-    family: Literal["KeypointDetectionYOLO"] = Field(
-        ..., description="Decoding family."
+    n_prototypes: Optional[int] = Field(
+        None, description="Number of prototypes per bbox in YOLO instance segmnetation."
     )
-    outputs: OutputsKeypointDetectionYOLO = Field(
-        description="A configuration specifying which output names from the `outputs` block of the archive are fed into the head."
+    n_keypoints: Optional[int] = Field(
+        None, description="Number of keypoints per bbox in YOLO keypoint detection."
+    )
+    is_softmax: Optional[bool] = Field(
+        None,
+        description="True, if output is already softmaxed in YOLO instance segmentation.",
     )
-    n_keypoints: int = Field(description="Number of keypoints per bbox.")
 
     @field_validator("family")
     def validate_label_type(
         cls,
         value,
     ):
-        if value != "KeypointDetectionYOLO":
+        if value not in [
+            "ObjectDetectionYOLO",
+            "InstanceSegmentationYOLO",
+            "KeypointDetectionYOLO",
+            "OBBDetectionYOLO",
+        ]:
             raise ValueError("Invalid family")
         return value
 
+    @model_validator(mode="before")
+    def validate_task_specific_fields(
+        cls,
+        values,
+    ):
+        family = values.get("family")
+
+        required_fields = {
+            "InstanceSegmentationYOLO": [
+                "postprocessor_path",
+                "n_prototypes",
+                "is_softmax",
+            ],
+            "KeypointDetectionYOLO": ["n_keypoints"],
+        }
+
+        unsupported_fields = {
```

Comment:
This invalidates instance segmentation with keypoints. We don't support training of this for now, so it's fine, just something to note.